src/test/java/bio/terra/service/snapshot/SnapshotConnectedTest.java

```diff
@@ -164,6 +164,9 @@ void testArrayStruct() throws Exception {
     // Verify that the parent id wasn't set on snapshot creation.
     verify(samService).createSnapshotResource(any(), eq(summaryModel.getId()), eq(null), any());
 
+    // Set the dataset snapshot count so that the snapshot count is accurate
+    datasetArraySummary.snapshotCount(1);
```

Comment:
This approach is hacky, but the core issue is that the dataset summary is no longer an immutable object, so after snapshots are ingested, this count changes. This causes our downstream asserts to fail due to the dataset summary on the snapshot and the original dataset summary being different.